### PR TITLE
[CWF][Operator] Add status reporting from CWF Operator to Redis

### PR DESCRIPTION
**Pull request by mpgermano**
Signed-off-by: mgermano <mgermano@fb.com>

## Summary

This PR adds status reporting for HA clusters from the CWF operator
to Redis. This state will then by synced to the orc8r, to allow for displaying
in the NMS

It's implementation consists of
-  status reporter component to the operator
- client wrapper for redis to properly store state

## Test Plan

Add unit test for the redis client.
Deploy to the lab and ensure that pipeline Operator -> Redis -> Orc8r works.

## Additional Information

- [x] This change is backwards-breaking

This change breaks the existing CRD spec for an HAcluster. To fix this, 
we need to re-create the CRD:
```
kubectl -n magma -f delete -f deploy/crds/magma.cwf.k8s_hacluster_crd.yaml
kubectl -n magma -f delete -f deploy/crds/magma.cwf.k8s_v1alpha1_hacluster_cr.yaml

Copy new files from `magma/cwf/k8s/deploy/crds`.
Update `magma.cwf.k8s_v1alpha1_hacluster_cr.yaml` to have 
the appropriate `GatewayResourceNames` and `haPairID` values

kubectl -n magma -f create -f deploy/crds/magma.cwf.k8s_hacluster_crd.yaml
kubectl -n magma -f create -f deploy/crds/magma.cwf.k8s_v1alpha1_hacluster_cr.yaml

```

Given that we're only running this in a lab, this shouldn't cause any major issues.

### CODE REVIEW

**Comment by mpgermano** [comment]
Wait to merge until https://github.com/magma/magma/pull/3018 lands. Will use that model and state type instead

File: orc8r/gateway/go/redis/client_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package redis
+
+import (
+	"testing"
+
+	"github.com/alicebob/miniredis"
```
Comment (emakeev) [inline orc8r/gateway/go/redis/client_test.go:19]: 👍

File: orc8r/gateway/go/redis/client.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+)
+
+type RedisStateClient struct {
+	redisClient *redis.Client
+	stateSerde  RedisStateSerde
+}
+
+// RedisStateSerde defines an interfaces for state serialization
+type RedisStateSerde interface {
+	Serialize(in interface{}, version uint64) ([]byte, error)
```
Comment (emakeev) [inline orc8r/gateway/go/redis/client.go:38]: Serialize may need some documentation, specifically - how version is used

**Review by emakeev — APPROVED** [review]
LGTM, some cosmetic suggestions inline

**Review by xjtian — APPROVED** [review]

File: cwf/k8s/cwf_operator/deploy/operator.yaml [hunk 3]
```diff
@@ -58,3 +58,6 @@ spec:
                   fieldPath: metadata.name
             - name: OPERATOR_NAME
               value: "cwf-operator"
+            # Replace value for redis server address to enable status reporting
+            # - name: REDIS_ADDR
+            #  value: localhost:6380
```
Comment (xjtian) [inline cwf/k8s/cwf_operator/deploy/operator.yaml:63]: ?

Comment (mpgermano) [reply cwf/k8s/cwf_operator/deploy/operator.yaml:63]: The redis-server address will be based on the redis helm deployment so default configuration doesn't really work

**Review by uri200 — APPROVED** [review]

**Pull request merged**